api/v1alpha2/bundledeployment_types.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,shortName={"bd","bds"}
+// BundleDeployment is the Schema for the bundledeployments API
+type BundleDeployment struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   BundleDeploymentSpec   `json:"spec"`
+	Status BundleDeploymentStatus `json:"status,omitempty"`
+}
+
+// BundleDeploymentSpec defines the desired state of BundleDeployment
+type BundleDeploymentSpec struct {
+	// Source configures how to pull the bundle content.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems:=1
+	Sources []BundleDeplopymentSource `json:"sources"`
+	// Format refers to the bundle type which is being passed through
+	// the bundle deployment API.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=plain;helm;registry
+	Format string `json:"format"`
```

Comment:
The discussion mentioned `Paused` field. I've left it out for now, since controller logic is not designed to be paused / or to skip reconcile of certain BDs. Do we want to add now, or in next iteration?